### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by chocol4te** [rse-standrewscs/shallow-water#9]
Currently, nearly all comments have been copied from the Fortran implementation. In most cases this is acceptable as the logic is the same so the comments are perfectly valid. 

However, the doc comments could be improved by providing descriptions for each module and a top level summary of the project.

**Comment by alex-konovalov** [comment]
Does Rust have a proper documenting tool, BTW?

**Comment by chocol4te** [comment]
Yes, `cargo doc` generates all documentation for a project ([guide here](https://doc.rust-lang.org/rust-by-example/meta/doc.html)). Currently CI is building and deploying documentation to [https://rse-standrewscs.github.io/shallow-water/shallow_water/index.html](https://rse-standrewscs.github.io/shallow-water/shallow_water/index.html), but once we have a stable release, we could have it on [docs.rs]().
